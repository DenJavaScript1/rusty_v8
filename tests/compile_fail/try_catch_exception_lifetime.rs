// Copyright 2019-2020 the Deno authors. All rights reserved. MIT license.
use rusty_v8 as v8;

pub fn main() {
  let mut isolate = v8::Isolate::new(mock());
  let mut try_catch = v8::TryCatch::new(&mut isolate);
  let try_catch = try_catch.enter();

  let _exception = {
    let mut hs = v8::HandleScope::new(&mut isolate);
    let hs = hs.enter();
    try_catch.exception(hs).unwrap()
  };
}

fn mock<T>() -> T {
  unimplemented!()
}
