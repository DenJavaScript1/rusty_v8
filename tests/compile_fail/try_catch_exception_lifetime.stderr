error[E0597]: `hs` does not live long enough
  --> $DIR/try_catch_exception_lifetime.rs:11:14
   |
9  |   let _exception = {
   |       ---------- borrow later stored here
10 |     let mut hs = v8::HandleScope::new(&mut isolate);
11 |     let hs = hs.enter();
   |              ^^ borrowed value does not live long enough
12 |     try_catch.exception(hs).unwrap()
13 |   };
   |   - `hs` dropped here while still borrowed
